lib/main.coffee

```diff
@@ -14,14 +33,36 @@ module.exports =
       (executablePath) =>
         @executablePath = executablePath
 
+    # Add a listener to reload this package if the custom grammar scopes setting
+    # changes. Grammar scopes can only be set at load time.
+    @subscriptions.add atom.config.observe 'linter-tidy.customGrammarScopes',
+      () ->
```

Comment:
The reason I reloaded the package on grammar scope config change was because I didn't know if the `grammarScopes` array in the `provider` object was kept by reference by the `provideLinter` consumer. Since the array is used directly and not copied by the consumer, I'll change it as you suggested.